persistence-modules/hibernate-annotations/src/test/java/com/baeldung/hibernate/creationupdatetimestamp/HibernateCreationUpdateTimestampIntegrationTest.java

```diff
@@ -0,0 +1,116 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.baeldung.hibernate.creationupdatetimestamp.model.Book;
+
+class HibernateCreationUpdateTimestampIntegrationTest {
+
+    private static SessionFactory sessionFactory;
+
+    private Session session;
+
+    @BeforeAll
+    public static void beforeTests() {
+        Configuration configuration = new Configuration().addAnnotatedClass(Book.class)
+          .setProperty("hibernate.dialect", H2Dialect.class.getName())
+          .setProperty("hibernate.connection.driver_class", Driver.class.getName())
+          .setProperty("hibernate.connection.url", "jdbc:h2:mem:test")
+          .setProperty("hibernate.connection.username", "sa")
+          .setProperty("hibernate.connection.password", "")
+          .setProperty("hibernate.hbm2ddl.auto", "update");
+
+        ServiceRegistry serviceRegistry = new StandardServiceRegistryBuilder().applySettings(configuration.getProperties())
+          .build();
+
+        sessionFactory = configuration.buildSessionFactory(serviceRegistry);
+    }
+
+    @Test
+    void whenCreatingEntity_ThenCreatedOnIsSet() {
+        session = sessionFactory.openSession();
+        session.beginTransaction();
+        Book book = new Book();
+
+        session.save(book);
+        session.getTransaction()
+          .commit();
+        session.close();
+
+        assertNotNull(book.getCreatedOn());
+    }
+
+    @Test
+    void whenCreatingEntity_ThenCreatedOnAndLastUpdatedOnAreBothSet() {
+        session = sessionFactory.openSession();
+        session.beginTransaction();
+        Book book = new Book();
+
+        session.save(book);
+        session.getTransaction()
+          .commit();
+        session.close();
+
+        assertNotNull(book.getCreatedOn());
+        assertNotNull(book.getLastUpdatedOn());
+    }
+
+    @Test
+    void whenCreatingEntity_ThenCreatedOnAndLastUpdatedOnAreNotEqual() {
+        session = sessionFactory.openSession();
+        session.beginTransaction();
+        Book book = new Book();
+
+        session.save(book);
+        session.getTransaction()
+          .commit();
+        session.close();
+
+        assertNotEquals(book.getCreatedOn(), book.getLastUpdatedOn());
+    }
+
+    @Test
+    void whenUpdatingEntity_ThenLastUpdatedOnIsUpdatedAndCreatedOnStaysTheSame() {
+        session = sessionFactory.openSession();
+        session.beginTransaction();
+        Book book = new Book();
+        session.save(book);
+        session.flush();
+        Instant createdOnAfterCreation = book.getCreatedOn();
+        Instant lastUpdatedOnAfterCreation = book.getLastUpdatedOn();
+
+        String newName = "newName";
+        book.setTitle(newName);
+        session.getTransaction()
+          .commit();
+        session.close();
+        Instant createdOnAfterUpdate = book.getCreatedOn();
+        Instant lastUpdatedOnAfterUpdate = book.getLastUpdatedOn();
+
+        assertEquals(book.getTitle(), newName);
```

Comment:
Swap the order of the arguments. Should be assertEquals(expected, actual)

Reply:
Done